### DISCUSSION TITLE
Is there constant-time support for all algorithms?

### DISCUSSION THREAD

**Discussion by derstrand** [open-quantum-safe/liboqs#1347] [Q&A]
Does liboqs support constant-time for all algorithms and for both platforms, x86 and aarch64?

**Comment by baentsch** [comment]
We strive for this property and have a test option for it (see [OQS_ENABLE_TEST_CONSTANT_TIME](https://github.com/open-quantum-safe/liboqs/wiki/Customizing-liboqs#oqs_enable_test_constant_time) and the [latest x64 test run log](https://github.com/open-quantum-safe/liboqs/actions/runs/3921260838)) but this should probably not be taken as a guarantee. Adding a test for aarch64 may be sensible: Would you see a benefit in that and possibly be willing to contribute that?

Edit/Add: Just see in [the driver code](https://github.com/open-quantum-safe/liboqs/blob/fe3cb02cb13b860dc8796731f9c574f817a7c71e/.github/workflows/weekly.yml#L13-L24) that some SPHINCS+ and ClassicMcEliece variants are exempted from the tests -- so it's probably safe to assume they don't exhibit this property.

**Comment by dstebila** [comment]
Take a look at the algorithm data sheets (the *.md files in https://github.com/open-quantum-safe/liboqs/tree/main/docs/algorithms) to see which algorithms' implementations on which platforms we do this testing for.
